sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -351,7 +351,7 @@ def truncate_attribute_values(attrs)
           return EMPTY_ATTRIBUTES if attrs.nil?
 
           attribute_length_limit = @span_limits.attribute_length_limit
-          attrs.each { |key, value| attrs[key] = OpenTelemetry::Common::Utilities.truncate(value, attribute_length_limit) } if attribute_length_limit
+          attrs.each { |key, value| attrs[key] = OpenTelemetry::Common::Utilities.truncate_attribute_value(value, attribute_length_limit) if attribute_length_limit && value.is_a?(String) }
```

Comment:
This is getting more complicated than a one-liner is worth.
```suggestion
          return attrs if attribute_length_limit.nil?

          attrs.each do |key, value|
            attrs[key] = OpenTelemetry::Common::Utilities.truncate_attribute_value(value, attribute_length_limit) if value.is_a?(String)
          end
```